### PR TITLE
app-text/apvlv: update to 0.4.0

### PR DESCRIPTION
**Pull request by gbeliako**
Signed-off-by: Chaquen Beliakov Amaya <c@cgps.ch>

### CODE REVIEW

File: app-text/apvlv/apvlv-0.4.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit xdg cmake
```
Comment (juippis) [inline app-text/apvlv/apvlv-0.4.0.ebuild:5]: Empty line between please.
```suggestion
EAPI=8

inherit xdg cmake
```

File: app-text/apvlv/apvlv-0.4.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+DESCRIPTION="apvlv is a PDF/EPUB Viewer and its behaviour like Vim"
```
Comment (juippis) [inline app-text/apvlv/apvlv-0.4.0.ebuild:7]: We don't usually repeat package name in DESCRIPTION. This looks better:
`SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "apvlv - Alf's PDF/DJVU/EPUB Viewer like Vim")`
```suggestion
DESCRIPTION="Alf's PDF/DJVU/EPUB Viewer like Vim"
```
which is also closer to the current description.

File: app-text/apvlv/apvlv-0.4.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="debug djvu"
+
+DEPEND="
+	>=app-text/poppler-0.5.0[cairo,xpdf-headers(+)]
+	dev-libs/glib:2
+	x11-libs/gtk+:3
+	net-libs/webkit-gtk
+	app-text/ebook-tools
+	djvu? ( app-text/djvu:= )
```
Comment (juippis) [inline app-text/apvlv/apvlv-0.4.0.ebuild:22]: Looking at https://github.com/naihe2010/apvlv/blob/master/src/CMakeLists.txt you seem to be missing at least freetype as a dep.

File: app-text/apvlv/apvlv-0.4.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+virtual/pkgconfig
```
Comment (juippis) [inline app-text/apvlv/apvlv-0.4.0.ebuild:26]:
```suggestion
	virtual/pkgconfig
```

File: app-text/apvlv/apvlv-0.4.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+inherit xdg cmake
+
+DESCRIPTION="apvlv is a PDF/EPUB Viewer and its behaviour like Vim"
+HOMEPAGE="https://github.com/naihe2010/apvlv/"
+SRC_URI="https://github.com/naihe2010/apvlv/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug djvu"
```
Comment (juippis) [inline app-text/apvlv/apvlv-0.4.0.ebuild:14]: These use flags aren't doing anything in this ebuild? Maybe you forgot to write the rest? :)

File: app-text/apvlv/apvlv-0.4.0.ebuild (hunk 3 above)
Comment (gbeliako) [reply app-text/apvlv/apvlv-0.4.0.ebuild:23]: freetype is already a dependency of app-text/poppler, which is already included. Including freetype might be redundant.

Comment (juippis) [reply app-text/apvlv/apvlv-0.4.0.ebuild:23]: We don't depend on indirect deps like that. Imagine a scenario where one day poppler dropped freetype as a dep, or it became a dependency via a use flag. It'd break apvlv. 

So always list direct dependencies.

File: app-text/apvlv/apvlv-0.4.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	cmake_src_configure
+}
+src_install() {
+	cmake_src_install
+	newicon -s 32 icons/pdf.png ${PN}.png
+}
+RDEPEND="${DEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+"
```
Comment (juippis) [inline app-text/apvlv/apvlv-0.4.0.ebuild:41]: Let's keep the default ebuild structure here, where dependencies and overall the global stuff comes before ebuild phases.
```suggestion
RDEPEND="
	app-text/ebook-tools
	>=app-text/poppler-0.5.0:=[cairo,xpdf-headers(+)]
	dev-libs/glib:2
	dev-libs/libxml2
	net-libs/webkit-gtk
	x11-libs/cairo
	x11-libs/gdk-pixbuf:2
	x11-libs/gtk+:3
	x11-libs/pango
	djvu? ( app-text/djvu:= )
"
DEPEND="${RDEPEND}
	app-text/ghostscript-gpl"
BDEPEND="virtual/pkgconfig"
"

src_configure() {
	local mycmakeargs=(
		-DAPVLV_WITH_DJVU=$(usex djvu)
		-DAPVLV_ENABLE_DEBUG=$(usex debug)
		-DAPVLV_WITH_TXT=ON
	)
	cmake_src_configure
}

src_install() {
	cmake_src_install
	newicon -s 32 icons/pdf.png ${PN}.png
}

```
I also noticed the freetype dep seems bogus, even though upstream claims so. Maybe only headers are needed (add to DEPEND-only)? Also ghostscript-gpl seems to be necessary to get this installed.

**Review by juippis — COMMENTED** [review]
It's working, but there are two problems:
```
 * QA Notice: The ebuild is installing to one or more unexpected paths:
 * 
 *   /usr/share/doc/apvlv
 * 
```
This is a bigger problem. It seems like it's just installing an empty directory there but the real contents are in a proper location.
```
/usr/share/doc
/usr/share/doc/apvlv
/usr/share/doc/apvlv-0.4.0
/usr/share/doc/apvlv-0.4.0/AUTHORS
/usr/share/doc/apvlv-0.4.0/NEWS.bz2
/usr/share/doc/apvlv-0.4.0/README.md.bz2
/usr/share/doc/apvlv-0.4.0/THANKS.bz2
/usr/share/doc/apvlv-0.4.0/TODO
/usr/share/doc/apvlv/Startup.pdf
/usr/share/doc/apvlv/apvlvrc.example.bz2
```
So just calling `rmdir "${ED}/usr/share/doc || die` in `src_install()` should be fine. Another issue is:
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/apvlv
```
Which may be solvable through the cmake configure section, but may require using a patch or sed to work properly. This is not **as** important as the above issue, but a nice bonus if you can try and figure it out.

File: app-text/apvlv/apvlv-0.4.0.ebuild (hunk 6 above)
Comment (gbeliako) [reply app-text/apvlv/apvlv-0.4.0.ebuild]: created a patch for the CXX flags and fixed the rest

**Review by juippis — APPROVED** [review]
Thanks a lot! Btw you don't have to become a maintainer for packages you contribute to, but happy to accept you if you really want to!

**Pull request closed without merging**